internal/component/postgresql/postgresdbaccess.go

```diff
@@ -55,18 +56,25 @@ type PostgresDBAccess struct {
 	metadata postgresMetadataStruct
 	db       pgxPoolConn
 
+	ensureTableFn func(context.Context, pgx.Tx, EnsureTableOptions) error
```

Comment:
Can you rename to "migrateFn" or something that indicates it is more than just ensuring a table?